### PR TITLE
node attribute macro and node creates

### PR DESCRIPTION
**Pull request by mrinalwadhwa**
chore(rust): add gitignore
feat(rust): ockam node attribute macro
feat(rust): no_std node
feat(rust): tokio node

based off #748, should be merged after that.

### CODE REVIEW

**Review by jared-s — APPROVED** [review]
:tada:

File: implementations/rust/ockam/src/queue.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+use crate::address::Addressable;
+use alloc::collections::VecDeque;
+
+pub trait Queue<T> {
+    fn enqueue(&mut self, element: T) -> crate::Result<bool>;
+    fn dequeue(&mut self) -> Option<T>;
+    fn is_empty(&self) -> bool;
+}
```
Comment (robinbudd) [inline implementations/rust/ockam/src/queue.rs:8]: I think all three of us (you, Jared and I) have all implemented some version of a queue!

**Review by robinbudd — APPROVED** [review]

File: implementations/rust/ockam/src/queue.rs (hunk 1 above)
Comment (mrinalwadhwa) [reply implementations/rust/ockam/src/queue.rs:8]: this code is in this PR only because my branch is based on Jared's queue PR  #748

Comment (mrinalwadhwa) [reply implementations/rust/ockam/src/queue.rs:8]: so one less queue 😄

**Pull request merged**